### DISCUSSION TITLE
give_set_error not displaying errors return in form

### DISCUSSION THREAD

**Discussion by ChelseaKaweb** [impress-org/givewp#6662] [Developer Q&A] [answered]
Below use of give hook to do custom validation

```
function validate_postcode($valid_data, $data)
{
    $forms = array(38450, 37893, 39199);

   if (!in_array($data['give-form-id'], $forms)) {
        return;
    }

    $postcode = ($data['postcode'] ? esc_html($data['postcode']) : '');

    if (empty($postcode)) {
        return;
    }

    $postcode_first = explode(' ', $postcode);
    $accepted_postcodes = array(
        'B23', 'B24', 'B43', 'B44', 'B72', 'B73', 'B74', 'B75', 'B76', 'B77', 'B78', 'B79', 'CV9', 'DE11', 'DE12', 'DE13', 'DE14', 'DE15', 'DE65'
    );
    $accepted_postcodes = array_map('strtolower', $accepted_postcodes);

    if (!in_array(strtolower($postcode_first[0]), $accepted_postcodes)) {
        give_set_error('give_message', __('Sorry, we are not collecting from your area.', 'give'));


    }
}

add_action('give_checkout_error_checks', 'validate_postcode', 10, 1);
```

The above seems to be working as it can be seen in the response call, however this validation output is not displaying anywhere on the frontend of the form.

![Screenshot 2023-01-03 at 16 30 55](https://user-images.githubusercontent.com/121884661/210399404-3e162a6e-99ae-4c31-bf7b-94acf91c3adf.png)
![Screenshot 2023-01-03 at 16 31 02](https://user-images.githubusercontent.com/121884661/210399421-551b67a0-d0fc-4009-a4d6-b08210593e00.png)

WP & Plugin is up to date

**Comment by kjohnson** [comment, marked as answer]
Hey @ChelseaKaweb, it looks like that is a bug, which I was able to reproduce  with the Classic Form Template.

So that you can be kept up to date with the status of this bug I would suggest that you open a [support ticket.](https://wordpress.org/support/plugin/give/) Our support team is best equipped to keep track of communicates with customers.

You can provide them with this reduced test case, which they can troubleshoot using the Classic Form Template.

```php
add_action('give_checkout_error_checks', function($valid_data, $data) {
   give_set_error('my_error', __('Sorry, we are not collecting from your area.', 'give'));
}, 10, 2);
```

**Reply by iamgracey** [reply]
hi, has been this resolved now?

**Comment by ChelseaKaweb** [comment]
Hi,

No it was just resolved by the looks of things https://wordpress.org/support/topic/custom-validation-not-displaying-feedback/ <https://wordpress.org/support/topic/custom-validation-not-displaying-feedback/>

Thanks
 <http://www.kaweb.co.uk/?utm_source=signaturesatori&utm_medium=email&utm_campaign=general_signature>	chelsea thomas-round
account manager / developer
w: kaweb.co.uk <https://kaweb.co.uk/?utm_source=signaturesatori&utm_medium=email&utm_campaign=general_signature> | t: 01543 414564 <tel:01543414564>
Sweet Digital & Click are brands of Kaweb, an ISO 27001 certified company.

This email and any attachments to it may be confidential and are intended solely for the use of the individual to whom it is addressed. Any views or opinions expressed are solely those of the author and do not necessarily represent those of Kaweb Limited (Company No. 3891‌664. VAT Registration No. 747458689). If you are not the intended recipient of this email, you must neither take any action based upon its contents, nor copy or show it to anyone. Please contact the sender if you believe you have received this email in error.

**Reply by iamgracey** [reply]
is there a way to show the custom error message in the form after submitting?

**Comment by ChelseaKaweb** [comment]
Hi

Was there any progress with this?


https://feedback.givewp.com/bug-reports/p/givewp-should-display-custom-errors-correctly-on-the-classic-and-multi-step-form￼
GiveWP should display custom errors correctly on the classic and multi-step forms | Bug Reports | GiveWP Fundraising Plugin
feedback.givewp.com

Thank you
 <http://www.kaweb.co.uk/?utm_source=signaturesatori&utm_medium=email&utm_campaign=general_signature>	chelsea thomas-round 
developer
w: kaweb.co.uk <https://kaweb.co.uk/?utm_source=signaturesatori&utm_medium=email&utm_campaign=general_signature> | t: 01543 414564 <tel:01543414564>
Sweet Digital & Click are brands of Kaweb, an ISO 27001 certified company.

This email and any attachments to it may be confidential and are intended solely for the use of the individual to whom it is addressed. Any views or opinions expressed are solely those of the author and do not necessarily represent those of Kaweb Limited (Company No. 3891‌664. VAT Registration No. 747458689). If you are not the intended recipient of this email, you must neither take any action based upon its contents, nor copy or show it to anyone. Please contact the sender if you believe you have received this email in error.
